src/SuperstateToken.sol

```diff
@@ -604,18 +603,37 @@ contract SuperstateToken is ISuperstateToken, ERC20Upgradeable, PausableUpgradea
             to: supportedStablecoins[stablecoin].sweepDestination,
             value: inAmount
         });
-        _mintLogic({dst: msg.sender, amount: superstateTokenOutAmount});
+        _mintLogic({dst: to, amount: superstateTokenOutAmount});
 
-        //@TODO: use SubscribeV2 with to address
-        emit Subscribe({
+        emit SubscribeV2({
             subscriber: msg.sender,
+            to: to,
             stablecoin: stablecoin,
             stablecoinInAmount: inAmount,
             stablecoinInAmountAfterFee: stablecoinInAmountAfterFee,
             superstateTokenOutAmount: superstateTokenOutAmount
         });
     }
 
+    /**
+     * @notice The ```subscribeTo``` function takes in stablecoins and mints SuperstateToken in the proper amount for the to address depending on the current Net Asset Value per Share.
+     * @param to The address where USTB will be deposited at
+     * @param inAmount The amount of the stablecoin in
+     * @param stablecoin The address of the stablecoin to calculate with
+     */
+    function subscribeTo(address to, uint256 inAmount, address stablecoin) external {
```

Comment:
this can also just be called subscribe

Reply:
https://github.com/superstateinc/ustb/pull/88/commits/ceb54ea1855779cf55b5fdd89f7c7a09dfc28783